api/gateway-operator/v2beta1/controlplane_types.go

```diff
@@ -65,6 +65,7 @@ type ControlPlaneList struct {
 // ControlPlaneSpec defines the desired state of ControlPlane
 //
 // +apireference:kgo:include
+// +kubebuilder:validation:XValidation:message="",rule="self.dataplane.type != 'ref' || has(self.ingressClass)"
```

Comment:
```suggestion
// +kubebuilder:validation:XValidation:message="",rule="self.dataplane.type != 'ref' && has(self.ingressClass)"
```

Shouldn't this be `&&`?
Also, can we add a message here?

Reply:
If the type is set to something else than `ref` then the condition exits early (not checking the `self.ingressClass`.

If it's `ref` (then the first part does not equal to `true`) then we check the second part and enforce the ingress class being set.

---

Adding the message in 94d73c6e6f9e0e89271775b2b49c64b005b6cc42.